File: backend/rxas400adm-operation/src/main/java/com/rxas400adm/operation/domain/Operation.java
package com.rxas400adm.operation.domain;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
@TableName("rx_operation")
public class Operation {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String operationType;
    private String status;
    private String currentStep;
    private String targetType;
    private String targetName;
    private String requestData;
    private String resultData;
    private String errorCode;
    private String errorMessage;
    private Integer retryCount;
    private Integer maxRetry;
    private String idempotencyKey;
    private String riskLevel;
    private String requestedBy;
    private LocalDateTime requestedAt;
    private LocalDateTime startedAt;
    private LocalDateTime completedAt;
    private LocalDateTime updatedAt;
    private Integer version;
}

File: frontend/src/api/operation.ts
import request from './request'
import type { PaginatedResult } from './types'

export interface CreateOperationReq {
  operationType: string
  targetType?: string
  targetName?: string
  requestData?: string
  idempotencyKey?: string
}

export interface OperationVO {
  id: number
  operationType: string
  status: string
  currentStep?: string
  steps?: string[]
  targetType?: string
  targetName?: string
  requestData?: string
  resultData?: string
  errorCode?: string
  errorMessage?: string
  retryCount: number
  maxRetry: number
  idempotencyKey?: string
  riskLevel?: string
  requestedBy?: string
  requestedAt?: string
  startedAt?: string
  completedAt?: string
}

export function createOperation(data: CreateOperationReq): Promise<OperationVO> {
  return request.post('/operations', data)
}

export function getOperation(id: number): Promise<OperationVO> {
  return request.get(`/operations/${id}`)
}

export function listOperations(current: number, size: number): Promise<PaginatedResult<OperationVO>> {
  return request.get('/operations', { params: { current, size } })
}

export function retryOperation(id: number): Promise<OperationVO> {
  return request.post(`/operations/${id}/retry`)
}

export function cancelOperation(id: number): Promise<void> {
  return request.post(`/operations/${id}/cancel`)
}